### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by rbren**
CC @li-boxuan 

A couple changes here:
* Make sure mount path respects the user's environment when regenerating (important for my env)
* Exit whenever codeact asks for more input

### CODE REVIEW

File: tests/integration/mock/CodeActAgent/test_edits/prompt_002.log [hunk 1]
```diff
@@ -220,19 +220,16 @@ Fix typos in bad.txt. Do not ask me for confirmation at any point.
...
 ----------
 
 OBSERVATION:
-[File: /workspace/bad.txt (4 lines total)]
-1:This is a stupid typoo.
-2:Really?
-3:No mor typos!
-4:Enjoy!
+[File: /workspace/bad.txt (1 lines total)]
+1:ENVIRONMENT REMINDER: You have 8 turns left to complete the task.
```
Comment (li-boxuan) [inline tests/integration/mock/CodeActAgent/test_edits/prompt_002.log:232]: this looks very suspicious.

Comment (rbren) [reply tests/integration/mock/CodeActAgent/test_edits/prompt_002.log:232]: agree! can't figure it out...

Comment (neubig) [reply tests/integration/mock/CodeActAgent/test_edits/prompt_002.log:232]: @rbren : I think I fixed this when I fixed the copy command in the other PR.

Comment (rbren) [reply tests/integration/mock/CodeActAgent/test_edits/prompt_002.log]: 🤔 I think I have your changes reflected here, no?

Comment (neubig) [reply tests/integration/mock/CodeActAgent/test_edits/prompt_002.log]: Ah, I see. Did you delete the content of `tests/integration/mock/CodeActAgent/test_edits/` and re-generate it?

Comment (rbren) [reply tests/integration/mock/CodeActAgent/test_edits/prompt_002.log]: something really weird was going on in my env. It was mounting the wrong directory into the sandbox container--a ghost directory I can't find anywhere on my machine 👻

**Review by neubig — APPROVED** [review]
Looks good, thanks!

File: tests/integration/regenerate.sh [hunk 2]
```diff
@@ -44,26 +53,38 @@ for ((i = 0; i < num_of_tests; i++)); do
...
+      AGENT=$agent \
       poetry run pytest -s ./tests/integration/test_agent.py::$test_name
     TEST_STATUS=$?
     # Re-enable 'exit on error'
     set -e
 
     if [[ $TEST_STATUS -ne 0 ]]; then
       echo -e "\n\n\n\n========$test_name failed, regenerating test data for $agent========\n\n\n\n"
-      # trick: let's not clean up $WORKSPACE_BASE folder, which might contain the
-      # artifacts we need that are auto-gerated by the test
```
Comment (li-boxuan) [inline tests/integration/regenerate.sh:58]: Oh I love to have this trick/hack removed.

File: tests/integration/regenerate.sh [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+  WORKSPACE_BASE=$(pwd)
+fi
+
+WORKSPACE_MOUNT_PATH+="/_test_workspace"
+WORKSPACE_BASE+="/_test_workspace"
+
 SANDBOX_TYPE="ssh"
 MAX_ITERATIONS=10
 
-# FIXME: SWEAgent hangs, so it goes last
```
Comment (li-boxuan) [inline tests/integration/regenerate.sh:9]: @rbren Is there any specific reason for removal of L9?

**Pull request merged**